iwa-console/MSAL/MSALClientHelper.cs

```diff
@@ -0,0 +1,270 @@
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MSALClientHelper"/> class.
+        /// </summary>
+        public MSALClientHelper(AzureADConfig azureADConfig)
+        {
+            this.AzureADConfig = azureADConfig;
+
+            this.InitializePublicClientApplicationBuilder();
+        }
+
+        /// <summary>
+        /// Initializes the MSAL's PublicClientApplication builder.
+        /// </summary>
+        /// <autogeneratedoc />
+        private void InitializePublicClientApplicationBuilder()
+        {
+            this.PublicClientApplicationBuilder = PublicClientApplicationBuilder.Create(AzureADConfig.ClientId)
+                .WithAuthority(string.Format(AzureADConfig.Authority, AzureADConfig.TenantId))
+                .WithRedirectUri(string.Format(AzureADConfig.RedirectURI, AzureADConfig.ClientId))      // Skipping this will make MSAL fall back to older Uri: urn:ietf:wg:oauth:2.0:oob
+                .WithLogging(new IdentityLogger(EventLogLevel.Warning), enablePiiLogging: false)        // This is the currently recommended way to log MSAL message. For more info refer to https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/wiki/logging. Set Identity Logging level to Warning which is a middle ground
+                .WithClientCapabilities(new string[] { "cp1" });                                        // declare this client app capable of receiving CAE events- https://aka.ms/clientcae
+        }
+
+        /// <summary>
+        /// Initializes the public client application of MSAL.NET with the required information to correctly authenticate the user.
+        /// </summary>
+        /// <returns></returns>
+        public async Task<IAccount> InitializePublicClientAppAsync()
+        {
+            // Initialize the MSAL library by building a public client application
+            this.PublicClientApplication = this.PublicClientApplicationBuilder.Build();
+
+            await AttachTokenCache();
+            return await FetchSignedInUserFromCache().ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Initializes the public client application of MSAL.NET with the required information to correctly authenticate the user.
+        /// </summary>
+        /// <returns>An IAccount of an already signed-in user (if available)</returns>
+        public async Task<IAccount> InitializePublicClientAppForWAMBrokerAsync(IntPtr? handle)
+        {
+            // Initialize the MSAL library by building a public client application for authenticating using WAM
+            this.PublicClientApplication = this.PublicClientApplicationBuilder
+                    .WithBrokerPreview(true)
+                    .WithParentActivityOrWindow(() => { return handle.Value; })                                     // Specify Window handle - (required for WAM).
+                    .Build();
+
+            this.IsBrokerInitialized = true;
```

Comment:
This is bad design, this flag can be set depending on the `Intialize` method used. Please remove flag and `InitializePublicClientAppAsync`

Reply:
Based on what I see there is no signature that only takes an `IntPtr`. Needs to be wrapped in a function.